src/left-nav.jsx

```diff
@@ -225,6 +236,28 @@ const LeftNav = React.createClass({
       </div>
     );
   },
+  
+  _isOpen() {
+    return this.props.open // component is controlled
+        // component is uncontrolled and the state indicates that it's open
+        || (this.props.open === undefined && this.state.open);
+  },
+  
+  _shouldShow() {
+    return this._isOpen() || !!this.state.swiping;  // component is swiping;  
+  },
+  
+  _close() {
+    if (this.props.open === undefined) this.setState({ open: false });
+    if (this.props.onChangeRequest) this.props.onChangeRequest(false);
```

Comment:
Well... then I should un-deprecate onNavClose and onNavOpen too right? :sweat_smile: 
This was simply a fallback for when users want to leave it uncontrolled yet still be notified of the changes, I deprecated onNavClose and onNavOpen, you think it's best if they stay around and turn this into: 

```js
if (this.props.open === undefined) this.setState({ open: false });
else if (this.props.onChangeRequest) this.props.onChangeRequest(false);
```